Lib/test/libregrtest/save_env.py

```diff
@@ -292,6 +299,25 @@ def restore_warnings_showwarning(self, fxn):
         warnings = self.get_module('warnings')
         warnings.showwarning = fxn
 
+    def get_stty_echo(self):
+        termios = self.try_get_module('termios', demand=True)
+        if not os.isatty(fd := sys.__stdin__.fileno()):
+            return None
+        attrs = termios.tcgetattr(fd)
+        lflags = attrs[3]
+        return bool(lflags & termios.ECHO)
+
```

Comment:
Nit: the style in here is to squash the `get_*` and `restore_*` functions together to mark them as a couple:
```suggestion
```